### PR TITLE
Support encryptor allowlist

### LINKED ISSUE #305: Support encryptor allowlist
- Add a `registerEncrypter(provider, signer, ritualId, addresses)` helper method to `@nucypher/taco`
- Uses authority account `signer` to authorize encrypter accounts on the `GlobalAllowList`

### LINKED ISSUE #324: Expose `registerEncrypters` from `taco` API

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:** 
- 2

**What this does:**
> High-level idea of the changes introduced in this PR. 
> List relevant API changes (if any), as well as related PRs and issues.

**Issues fixed/closed:**
- Fixes #305
- Fixes #324

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
- Replacing `string` usage with `ChecksumAddress` usage where possible
- Are there any other facilities that we should provide to make encrypter onboarding easier?
- Supplemented by `isAuthorizedEncrypter` added in #301 
- `ethers-typechain` directory contains auto-generated code and need not be reviewed
- Rebase & merge after implementing https://github.com/nucypher/nucypher-ts/issues/310

### CODE REVIEW

File: packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+/* eslint-disable */
+
+import { Contract, Signer, utils } from "ethers";
+import type { Provider } from "@ethersproject/providers";
+import type {
+  GlobalAllowList,
+  GlobalAllowListInterface,
+} from "../GlobalAllowList";
+
+const _abi = [
```
Comment (theref) [inline packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts:12]: is there a difference between this `_abi` and `packages/shared/abis/GlobalAllowList.json`?

**Review by KPrasch — COMMENTED** [review]
Looking good, although if there's an incoming PR that will integrate `nucypher-contracts` `v0.5.0` do we want to wait for the first?

File: packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts (hunk 1 above)
Comment (piotr-roslaniec) [reply packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts:12]: This is code generated by `typechain`, the difference includes generated (inferred) types.

**Comment by piotr-roslaniec** [comment]
> (...) there's an incoming PR that will integrate nucypher-contracts v0.5.0 do we want to wait for that first?

Yes, I will add this information to the PR description.

File: packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts (hunk 1 above)
Comment (theref) [reply packages/shared/src/contracts/ethers-typechain/factories/GlobalAllowList__factory.ts:12]: ok cool, so as one changes, the other will be automatically kept up to date by typechain

File: packages/shared/src/contracts/registry.ts [hunk 2]
```diff
@@ -4,33 +4,38 @@ import { ChainId } from '../web3';
...
 const POLYGON: Contracts = {
   SUBSCRIPTION_MANAGER: '0xB0194073421192F6Cf38d72c791Be8729721A0b3',
   COORDINATOR: undefined,
+  GLOBAL_ALLOW_LIST: undefined,
 };
 
 const MUMBAI: Contracts = {
   SUBSCRIPTION_MANAGER: '0xb9015d7b35ce7c81dde38ef7136baa3b1044f313',
   COORDINATOR: '0x8E49989F9D3aD89c8ab0de21FbA2E00C67ca872F',
+  GLOBAL_ALLOW_LIST: '0x7b521E78CFaf55fa01433181d1D636E7e4b73243',
```
Comment (derekpierre) [inline packages/shared/src/contracts/registry.ts:19]: We should be careful here. Both `tapir` and `lynx` use Mumbai but have different GlobalAllowList contracts:
- Lynx: `0x7b521E78CFaf55fa01433181d1D636E7e4b73243`
- Tapir: `0x5Ff235481bbDBF5dFCC908d987506FA6a8696678`

How would this resolution work currently? Keying by "domain" is better than solely by blockchain/chain_id.

**Review by derekpierre — CHANGES_REQUESTED** [review]

File: packages/shared/src/contracts/registry.ts (hunk 2 above)
Comment (piotr-roslaniec) [reply packages/shared/src/contracts/registry.ts:19]: Will address post https://github.com/nucypher/nucypher-ts/pull/314 changes

**Pull request closed without merging**